lib/chef/provider/user/dscl.rb

```diff
@@ -628,15 +754,19 @@ def mac_osx_version_10_7?
         end
 
         def mac_osx_version_less_than_10_7?
-          versions = mac_osx_version.split(".")
-          # Make integer comparison in order not to report 10.10 less than 10.7
-          (versions[0].to_i <= 10 && versions[1].to_i < 7)
+          major, minor = mac_osx_major_and_minor_versions
+
+          major <= 10 && minor < 7
         end
 
         def mac_osx_version_greater_than_10_7?
-          versions = mac_osx_version.split(".")
-          # Make integer comparison in order not to report 10.10 less than 10.7
-          (versions[0].to_i >= 10 && versions[1].to_i > 7)
+          major, minor = mac_osx_major_and_minor_versions
+
+          major >= 10 && minor > 7
```

Comment:
This will break for OSX `11.0` but I don't know how realistic of an expectation that is.  You could use `(major = 10 && minor > 7) || major > 10` to be safe.